scripts/rddepman.ts

```diff
@@ -0,0 +1,197 @@
+import * as tools from 'scripts/dependencies/tools';
+import { WSLDistro, HostResolverHost } from 'scripts/dependencies/wsl';
+import {
+  DependencyVersions, readDependencyVersions, writeDependencyVersions, Dependency, AlpineLimaISOVersion, getOctokit,
+} from 'scripts/lib/dependencies';
+
+const MAIN_BRANCH = 'main';
+const GITHUB_OWNER = process.env.GITHUB_OWNER || 'rancher-sandbox';
+const GITHUB_REPO = process.env.GITHUB_REPO || 'rancher-desktop';
+
+type VersionComparison = {
+  name: string;
+  currentVersion: string | AlpineLimaISOVersion;
+  latestVersion: string | AlpineLimaISOVersion;
+};
+
+const dependencies: Dependency[] = [
+  new tools.KuberlrAndKubectl(),
+  new tools.Helm(),
+  new tools.DockerCLI(),
+  new tools.DockerBuildx(),
+  new tools.DockerCompose(),
+  new tools.DockerProvidedCredHelpers(),
+  new tools.Trivy(),
+  new tools.GuestAgent(),
+  new tools.Steve(),
+  new tools.RancherDashboard(),
+  new tools.ECRCredHelper(),
+  new LimaAndQemu(),
+  new AlpineLimaISO(),
+  new WSLDistro(),
+  new HostResolverHost(), // we only need one of HostResolverHost and HostResolverPeer
+  new MobyOpenAPISpec(),
+];
+
+function git(...args: string[]): number | null {
+  const result = spawnSync('git', args);
+
+  if (result.error) {
+    throw result.error;
+  }
+
+  return result.status;
+}
+
+function printable(version: string | AlpineLimaISOVersion): string {
+  return typeof version === 'string' ? version : version.isoVersion;
+}
+
+function getBranchName(name: string): string {
+  return `rddepman/${ name }`;
+}
+
+function getTitle(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): string {
+  return `rddepman: bump ${ name } from ${ printable(currentVersion) } to ${ printable(latestVersion) }`;
+}
+
+function compareVersions(version1: string | AlpineLimaISOVersion, version2: string | AlpineLimaISOVersion): boolean {
+  if (typeof version1 === 'string' && typeof version2 === 'string') {
+    return version1 === version2;
+  } else if (typeof version1 !== 'string' && typeof version2 !== 'string') {
+    return version1.isoVersion === version2.isoVersion && version1.alpineVersion === version2.alpineVersion;
+  }
+  throw new Error('Types of version1 and version2 differ.');
+}
+
+async function createDependencyBumpPR(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): Promise<void> {
+  const title = getTitle(name, currentVersion, latestVersion);
+  const branchName = getBranchName(name);
+
+  console.log(`Creating PR "${ title }".`);
+  await getOctokit().rest.pulls.create({
+    owner: GITHUB_OWNER,
+    repo:  GITHUB_REPO,
+    title,
+    base:  MAIN_BRANCH,
+    head:  branchName,
+  });
+}
+
+async function getPulls(...options: Parameters<Octokit['rest']['pulls']['list']>): Promise<Awaited<ReturnType<Octokit['rest']['pulls']['list']>>['data']> {
```

Comment:
Maybe factor out `Octokit['rest']['pulls']['list']`, because it looks ugly and is repeated three times?
```ts
type listFn = Octokit['rest']['pulls']['list'];
```